docs/FAQ.rst

```diff
@@ -167,6 +167,16 @@ LightGBM
 
 --------------
 
+-  **Question 12**: Why is early stopping not enabled by default in LightGBM?
+
+-  **Solution 12**: Early stopping involves choosing a validation set, a special type of holdout which is used to evaluate the current state of the model after each iteration to see if training can stop.
+
+   In ``LightGBM``, `we have decided to require that users specify this set directly <./Parameters.html#valid>`_. Many options exist for splitting training data into training, test, and validation sets.
```

Comment:
```suggestion
   In ``LightGBM``, `we have decided to require that users specify this set directly <./Parameters.rst#valid>`_. Many options exist for splitting training data into training, test, and validation sets.
```

Reply:
@jameslamb Can you please approve the suggestion?

Reply:
@StrikerRUS Sorry for the delay. I disagree with it. It needs to be `.html` so that the link will work correctly.

See, for example, how all the links on the site are to specific `.html` files: https://lightgbm.readthedocs.io/en/latest/Python-Intro.html

`sphinx` will always take `foo.rst` and create a `foo.html`, so we can rely on the `.html` being there after running `sphinx-build`.

Reply:
@jameslamb There was a long story behind this. In short, we came to the agreement about that we need correct links on both GitHub and RTD. `.rst` extension allows to navigate correctly on GitHub, then we have a JS script which replaces `.rst` with `.html` after Sphinx job:
https://github.com/microsoft/LightGBM/blob/274f340ae8232c707428d1747a857f4fc2511146/docs/_static/js/script.js#L2-L3

Your link is working only for RTD. Take a look at any other docs for examples.

Reply:
Oh wow ok! Sorry, I had not noticed that before and wasn't aware we were doing that before. Ok awesome!

Thanks for the clarification.

Reply:
change committed

Reply:
@jameslamb No problem! I tried to describe all details :-)